sdk/src/main/java/com/mapbox/maps/Style.kt

```diff
@@ -20,15 +20,19 @@ import java.nio.ByteBuffer
  * @property pixelRatio the scale ratio of the style, default the device pixel ratio
  */
 class Style internal constructor(
-  private val styleManagerRef: StyleManagerInterface,
+  private var styleManager: StyleManagerInterface?,
   override val pixelRatio: Float
 ) : StyleInterface {
 
+  internal fun onDestroy() {
+    styleManager = null
+  }
```

Comment:
When is the style destroyed? seems currently it's destroyed when the MapboxMap is destroyed. I think to make the style object's lifecycle separate from MapboxMap's lifecycle, we could probably destroy the previous style when a new style is loaded? But I think it would be an behavioural breaking change, and many users might keep a reference to the style object for reuse, it could break their code.

Reply:
Yes, we need to destroy the previous style when a new style is loaded.